File: __Resources/Week 3/TestApp/TestApp/Post.m
//
//  Post.m
//  ToDoApp
//
//  Created by Phil Wright on 6/6/16.
//  Copyright © 2016 Touchopia, LLC. All rights reserved.
//

#import "Post.h"

@implementation Post

-(instancetype)initWithDictionary:(NSDictionary *)dict {
    if ( self = [super init] ) {
        self.title = [dict objectForKey:@"title"];
        self.body = [dict objectForKey:@"body"];
        self.userId = [dict objectForKey:@"userId"];
        self.postId = [dict objectForKey:@"id"];
    }
    return self;
}

@end

File: 46-Firebase2/SimpleFirebaseDatabase/SimpleFirebase/Post.swift
//
//  Post.swift
//  SimpleFirebase
//
//  Created by Phil Wright on 7/20/16.
//  Copyright © 2016 Touchopia, LLC. All rights reserved.
//

import UIKit
import Firebase
import FirebaseDatabase

class Post : NSObject {
    
    let ref = FIRDatabase.database().reference().child("posts")
    
    var key : String = ""
    var uid : String = ""
    var title : String = ""
    var body : String = ""
    
    override init() {
        
    }
    
    init(key: String, dict: Dictionary<String, AnyObject>) {
        
        self.key = key
        
        if let title = dict["title"] as? String {
            self.title = title
        }
        
        if let body = dict["body"] as? String {
            self.body = body
        }
    }
    
    func save() {
        
        let dict: [String:AnyObject] = [
            "title": self.title,
            "body": self.body
        ]
        
        let postRef = self.ref.childByAutoId()
        postRef.setValue(dict)
    }
}
